Formula/g/gupnp-tools.rb

```diff
@@ -31,7 +31,11 @@ class GupnpTools < Formula
   depends_on "libxml2"
 
   on_macos do
+    depends_on "at-spi2-core"
+    depends_on "cairo"
     depends_on "gettext"
+    depends_on "harfbuzz"
+    depends_on "pango"
   end
```

Comment:
Don't see linkage to most of these
```
==> brew linkage --cached gupnp-tools
System libraries:
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /opt/homebrew/opt/gettext/lib/libintl.8.dylib (gettext)
  /opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /opt/homebrew/opt/gssdp/lib/libgssdp-1.6.0.dylib (gssdp)
  /opt/homebrew/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtksourceview4/lib/libgtksourceview-4.0.dylib (gtksourceview4)
  /opt/homebrew/opt/gupnp/lib/libgupnp-1.6.0.dylib (gupnp)
  /opt/homebrew/opt/gupnp-av/lib/libgupnp-av-1.0.3.dylib (gupnp-av)
  /opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (libsoup)
  /opt/homebrew/opt/libxml2/lib/libxml2.2.dylib (libxml2)
Dependencies with no linkage:
  at-spi2-core
```

Reply:
I really think it's time for a DSL improvement where we can specify if a dependency is necessary, but won't have linkage.

Reply:
looks like linkage reporting is not consistent either?

<img width="548" alt="image" src="https://github.com/user-attachments/assets/319b7ca5-3e66-4775-9b90-8d624b71f370">

Reply:
this is the linkage check in my local

```
$ brew linkage --cached gupnp-tools
System libraries:
  /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon
  /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
  /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /opt/homebrew/opt/at-spi2-core/lib/libatk-1.0.0.dylib (at-spi2-core)
  /opt/homebrew/opt/cairo/lib/libcairo-gobject.2.dylib (cairo)
  /opt/homebrew/opt/cairo/lib/libcairo.2.dylib (cairo)
  /opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /opt/homebrew/opt/gettext/lib/libintl.8.dylib (gettext)
  /opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgmodule-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /opt/homebrew/opt/gssdp/lib/libgssdp-1.6.0.dylib (gssdp)
  /opt/homebrew/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtksourceview4/lib/libgtksourceview-4.0.dylib (gtksourceview4)
  /opt/homebrew/opt/gupnp/lib/libgupnp-1.6.0.dylib (gupnp)
  /opt/homebrew/opt/gupnp-av/lib/libgupnp-av-1.0.3.dylib (gupnp-av)
  /opt/homebrew/opt/harfbuzz/lib/libharfbuzz.0.dylib (harfbuzz)
  /opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (libsoup)
  /opt/homebrew/opt/libxml2/lib/libxml2.2.dylib (libxml2)
  /opt/homebrew/opt/pango/lib/libpango-1.0.0.dylib (pango)
  /opt/homebrew/opt/pango/lib/libpangocairo-1.0.0.dylib (pango)
```

Reply:
Meson defaults to `-dead_strip_dylibs` so if they aren't showing up anymore in CI, then linkage may have been pruned out.

Note that Sonoma was originally using the same buggy `ld` that Ventura is currently using. So old Sonoma bottles will be overlinked. If you want to check linkage, you should either check Monterey bottles, or Sonoma bottles built with Xcode 15.3